Add deltaRCM metadata

mperignon opened the pull request:
Metadata for standalone component - empty provides and uses.

mdpiper commented:
Two items:

1. Would you please convert the parameters file from YAML to JSON? (Please keep the YAML file, though, for future use!) Although we're moving to YAML, JSON is still what we're using in WMT. 

    When doing the conversion, be careful; the JSON decoder in the WMT client is finicky. The file must contain an array of objects, with each object representing a single parameter.

1. The parameters `comp_status`, `model_output__site_prefix `, and `model_output__case_prefix` may not be necessary -- they're used in TopoFlow, but are not needed by WMT. (Unless you're using them in Man's model?) Check out HydroTrend's **parameters.json** file for a good example.

mperignon commented:
1. Added json file. I have all kinds of converters at this point! Formatting looks ok.

2a. Ah, thought comp status was WMT's. Removed.
2b and 2c. Yes, those came from Topoflow, but deltaRCM needed output file prefixes and those names were good enough.

--------------------------

Thanks for the link to the Python packaging docs. Really helpful! I managed to build the delta model from the bmi-python example in a couple of hours through trial and error but it needs to be cleaned up, get more setters and getters, and get broken up into more logical sub-packages. It installs correctly and works, though!
https://github.com/mperignon/bmi-python-delta

I'm working on the python version of the Parker codes right now. I'll make them into a buildable package but it probably makes no sense to put them into WMT as they are right now (being 1D and such), but it could be released along with a set of plotting functions as EKT.

mdpiper commented:
Excellent! Let's create another (secondary) WMT instance and include this model.
